unstructured/partition/epub.py

```diff
@@ -29,13 +37,49 @@ def partition_epub(
         If True, the output will include page breaks if the filetype supports it
     metadata_last_modified
         The last modified date for the document.
-
     """
-    return convert_and_partition_html(
-        source_format="epub",
-        filename=filename,
-        file=file,
-        include_page_breaks=include_page_breaks,
-        metadata_filename=metadata_filename,
-        metadata_last_modified=metadata_last_modified,
-    )
+    exactly_one(filename=filename, file=file)
+    if filename is None:
+        filename = ""
+
+    if file is not None:
+        with tempfile.NamedTemporaryFile(delete=False) as tmp:
+            tmp.write(file.read())
+            filename = tmp.name
+        last_modification_date = get_last_modified_date_from_file(file)
+    else:
+        last_modification_date = get_last_modified_date(filename)
+
+    book = epub.read_epub(filename)
+    html_items = [item for item in book.items if isinstance(item, epub.EpubHtml)]
+    toc_href_and_title = []
+    elements = []
+
+    for item in book.toc:
+        # Some toc items may be tuple of multiple items, but all have the same href
+        if isinstance(item, tuple):
+            toc_href_and_title.append((item[0].href.split("#")[0], item[0].title))
+        else:
+            toc_href_and_title.append((item.href.split("#")[0], item.title))
+
+    for item in html_items:
```

Comment:
Instead of constructing `html_items` in a list comprehension, can we do:

```python
for item in book.items:
    if not isinstance(item, epub.EpubHtml):
        continue
    <rest of stuff>
```

Also, what else is in `book.items` aside from `EpubHtml`? Is that anything we want to capture?

Reply:
There are `EpubLink`, `EpubImage`, `EpubNcx` (epub page navigation info) and `EpubItem` (which seems to just be fomatting/css info). Do we want to capture the images?